ui/plugin-system/src/runtime/time-series-queries.ts

```diff
@@ -14,36 +14,83 @@
 import { TimeSeriesQueryDefinition } from '@perses-dev/core';
 import { useQuery, useQueries } from '@tanstack/react-query';
 import { TimeSeriesQueryContext } from '../model';
+import { TimeSeriesQueryPlugin } from '../model';
+import { VariableStateMap } from './template-variables';
 import { useTemplateVariableValues } from './template-variables';
 import { useTimeRange } from './time-range';
 import { useDatasourceStore } from './datasources';
-import { usePlugin, usePluginRegistry } from './plugin-registry';
+import { usePlugin, usePluginRegistry, usePlugins } from './plugin-registry';
 
 export interface UseTimeSeriesQueryOptions {
   suggestedStepMs?: number;
 }
 
+/**
+ * Returns a serialized string of the current state of variable values.
+ */
+function getVariableValuesKey(v: VariableStateMap) {
+  return Object.values(v)
+    .map((v) => JSON.stringify(v.value))
```

Comment:
Curious: is this because variables could be an array value?

Reply:
Yes exactly